### DISCUSSION TITLE
[Bug]: Foreign key mismatch only when executed in browser

### DISCUSSION THREAD

**Discussion by wetbadger** [sqlitebrowser/sqlitebrowser#3165] [General]
### What did you do?

I entered this SQL code inside of sqlitebrowser:

CREATE TABLE IF NOT EXISTS places 
    ( 
    city TEXT,
    state TEXT NOT NULL,
    county TEXT,
    longitute REAL,
    latitude REAL,
    CONSTRAINT unq UNIQUE(city, state, county)
    );
CREATE TABLE IF NOT EXISTS people 
    ( 
    name TEXT UNIQUE,
    title TEXT, 
    placeId INT,
    CONSTRAINT fk_place FOREIGN KEY(placeId) REFERENCES places(rowid),
    CONSTRAINT unq UNIQUE(name)
    );

INSERT OR IGNORE INTO places 
    VALUES ('New York', 'NY', 'Kings', 40.7128, 74.0060);

INSERT OR IGNORE INTO people 
    VALUES ('James Johnson', 'Mr.', 1);

Then I pressed the "execute SQL" arrow button.

### What did you expect to see?

I expect not to see an error because this SQL will execute in sqlite3 from the command line.

The issue does not seem to be among the common causes listed here: https://www.sqlite.org/foreignkeys.html#fk_indexes

### What did you see instead?

Execution finished with errors.
Result: foreign key mismatch - "people" referencing "places"
At line 22:
INSERT OR IGNORE INTO people 
    VALUES ('James Johnson', 'Mr.', 1);

### DB4S Version

3.12.2

### What OS are you seeing the problem on?

Linux

### OS version

Mint 21

### Relevant log output

```shell
ROLLBACK TO SAVEPOINT "RESTOREPOINT";
PRAGMA database_list;
SELECT type,name,sql,tbl_name FROM "main".sqlite_master;
RELEASE "RESTOREPOINT";
PRAGMA foreign_keys = '1';
PRAGMA database_list;
SELECT type,name,sql,tbl_name FROM "main".sqlite_master;
PRAGMA encoding;
PRAGMA database_list;
SELECT type,name,sql,tbl_name FROM "main".sqlite_master;
SELECT COUNT(*) FROM "main"."people"
SELECT "_rowid_",* FROM "main"."people" LIMIT 0, 49999;
PRAGMA database_list;
SELECT type,name,sql,tbl_name FROM "main".sqlite_master;
SELECT COUNT(*) FROM "main"."people"
SELECT "_rowid_",* FROM "main"."people" LIMIT 0, 49999;
PRAGMA auto_vacuum;
PRAGMA automatic_index;
PRAGMA checkpoint_fullfsync;
PRAGMA foreign_keys;
PRAGMA fullfsync;
PRAGMA ignore_check_constraints;
PRAGMA journal_mode;
PRAGMA journal_size_limit;
PRAGMA locking_mode;
PRAGMA max_page_count;
PRAGMA page_size;
PRAGMA recursive_triggers;
PRAGMA secure_delete;
PRAGMA synchronous;
PRAGMA temp_store;
PRAGMA user_version;
PRAGMA wal_autocheckpoint;
SELECT 'x' NOT LIKE 'X';
PRAGMA database_list;
SELECT type,name,sql,tbl_name FROM "main".sqlite_master;
SELECT COUNT(*) FROM "main"."people"
SELECT "_rowid_",* FROM "main"."people" LIMIT 0, 49999;
PRAGMA database_list;
SELECT type,name,sql,tbl_name FROM "main".sqlite_master;
SELECT "_rowid_",* FROM "main"."places" LIMIT 0, 49999;
SELECT COUNT(*) FROM "main"."places"
PRAGMA database_list;
SELECT type,name,sql,tbl_name FROM "main".sqlite_master;
SELECT COUNT(*) FROM "main"."places"
SELECT "_rowid_",* FROM "main"."places" LIMIT 0, 49999;
```

### Prevention against duplicate issues

- [X] I have searched for similar issues

**Comment by chrisjlocke** [comment]
The error is from sqlite, not from DB4S, but I can't fathom where the error is.  
You say it works from the sqlite3 command-line.  What version of sqlite is that using?

**Comment by chrisjlocke** [comment]
Oh.  I lied.  You can't do this.

> The parent key must be a named column or columns in the parent table, not the [rowid](https://www.sqlite.org/lang_createtable.html#rowid).

This is why DB4S also doesn't allow you to select the rowId.

![image](https://user-images.githubusercontent.com/7872218/188989015-debe872d-3316-411f-bb0d-a1d42de0ea16.png)

**Comment by wetbadger** [comment]
I thought the entire purpose of __rowid__ was so you didn't have to make an id field.

**Comment by chrisjlocke** [comment]
"for the data in the underlying [B-tree](https://sqlite.org/fileformat2.html#btree) storage engine".  So it's for SQLite's use, not yours.  If you need to access or use that information, it's recommended to create a field.
